### ISSUE TITLE
how to train RGB image

### ISSUE THREAD

**Issue by 1244632315** [computational-cell-analytics/micro-sam#1028]
A brilliant job! Nowadays, I'm trying to apply this framework in my research domain. However, it seems that the current version didn't support RGB-based dataset. I want ensure whether I misunderstand the code or it's can't now.  Thx for your reply!!!

**Comment by 1244632315** [comment]
Thx for your quick reply. But does that mean I need to modify all the places involving data reading (before input the network, including _all the dataset and dataloader functions_) ? 
Because the current logic is concatating all the image to the channel: `c x (h, w) -> (c, h, w)`.

**Comment by anwai98** [comment]
For your RGB images, you need to prepare your input images in such a way that the RGB images are arranged in `(B, 3, Y, X)` format.

This should work with input images stored as `(Y, X, 3)` by switching `is_seg_dataset` to `False`, and passing the input images with corresponding masks (stored as `(Y, X)`)!

**Comment by 1244632315** [comment]
Thx, I'll try to solve it.

**Comment by anwai98** [comment]
Let us know if it works for you. Otherwise I can write a script for you if you share a few images and corresponding labels!

**Comment by 1244632315** [comment]
When I was ready to train RGB-based dataset. I found some errors during the training. Here, as the following figure shows, it should be `this_roi` rather than `roi` in th. Otherwise, there will be a assert error. 

![Image](https://github.com/user-attachments/assets/299d0276-6853-45e8-9eba-3d2403e1bc03)

![Image](https://github.com/user-attachments/assets/15b8c647-3154-4991-81b0-5d178346217f)

**Comment by anwai98** [comment]
Hi @1244632315,

The line of code you points to expects `this_roi` inside the `_get_paths` function, which is called later on by the primary `_load_image_collection` function.

I think it's best if you post your current `torch-em` based dataloader script here and I'll take a look at it and provide suggestions!
